### PR TITLE
Document more clearly the main limitations of the Hibernate Reactive extension

### PR DESCRIPTION
**Pull request by yrodiere**
See https://github.com/quarkusio/quarkus/issues/35485#issuecomment-1689833956

### CODE REVIEW

File: docs/src/main/asciidoc/hibernate-reactive.adoc [hunk 1]
```diff
@@ -283,10 +283,20 @@ know what is being done.
...
+* This extension cannot be used at the same time as Hibernate ORM.
+See https://github.com/quarkusio/quarkus/issues/13425.
+* This extension cannot be used at the same time as JDBC/Agroal.
+See https://github.com/quarkusio/quarkus/issues/33380.
++
+This also means this extension cannot be used at the same time as extensions that rely on JDBC,
+such as the Flyway extension.
+See https://github.com/quarkusio/quarkus/issues/10716.
+* Integration with the Envers extension is not supported.
+* Transaction demarcation cannot be done using `jakarta.transaction.Transactional`.
```
Comment (DavideD) [inline docs/src/main/asciidoc/hibernate-reactive.adoc:299]: I would add or mention the  `@WithTransaction` annoation as the alternative

**Review by DavideD — APPROVED** [review]
LGTM
Maybe @Sanne  should double check.

I left a comment about mentioning the `@WithTransaction` annotation as a substitute for `@Transactional`

File: docs/src/main/asciidoc/hibernate-reactive.adoc (hunk 1 above)
Comment (yrodiere) [reply docs/src/main/asciidoc/hibernate-reactive.adoc:299]: Will do

Comment (yrodiere) [reply docs/src/main/asciidoc/hibernate-reactive.adoc:299]: Actually that's specific to Hibernate Reactive with Panache, no?

Comment (yrodiere) [reply docs/src/main/asciidoc/hibernate-reactive.adoc]: It is. I mentioned it anyway, with links to the Hibernate Reactive with Panache guide.

**Comment by yrodiere** [comment]
Bump @Sanne :)

**Comment by DavideD** [comment]
I think we should merge it. If there are some tweaks to make, we can add them later

**Comment by yrodiere** [comment]
Yeah but I can't without an approval, add oddly it seems yours doesn't count (???)

**Review by DavideD — APPROVED** [review]

**Review by gsmet — APPROVED** [review]

**Pull request merged**